src/test/resources/examples/binary/controllers/ktor/Controllers.kt

```diff
@@ -0,0 +1,28 @@
+package ie.zalando.controllers
```

Comment:
It seems like the "authentication" test case isn't extended? I think that's an important element of generation that needs to be covered.

Reply:
Good point @atollk! 🔒 I have added the authentication test case for Ktor too.

The same commit also [adds a check](https://github.com/cjbooms/fabrikt/pull/280/commits/c82de2f89b0289d8459b7b300c5feb7b62777d2c#diff-e6e8af2766ab85eea84037212904553a47b13bcfa130967020f1a3b0ed1272d3R157) to see if `ControllerCodeGenOptionType.AUTHENTICATION` is enabled and only in that case generate auth code.

Reply:
Maybe I am missing something but it looks to me like it doesn't actually generate any authentication code, does it? I would expect the logic from https://ktor.io/docs/server-auth.html to be present in a generated controller, i.e. perform auth checks on relevant routes and extract the `Principal` object and pass it to the function stubs.

Reply:
You can actually just use `call.principal<Principal>()` and leave the specific cast to the user implementation. That's how I do it in my code base at least and so far I haven't encountered any issue with it.

Reply:
Amazing 🙌 This has been added in https://github.com/cjbooms/fabrikt/pull/280/commits/6535f06e7d7561563c6af25f508d0b8a874594d9.